### PR TITLE
networkmanager: Strict Content-Security-Policy

### PR DESCRIPTION
**Pull request by stefwalter**
Remove all inline and unsafe styling and scripts from networkmanager component. That also means removing bootstrap-select

### CODE REVIEW

File: pkg/networkmanager/interfaces.js [hunk 1]
```diff
@@ -1331,7 +1331,9 @@ PageNetworking.prototype = {
         function highlight_netdev_row(event, id) {
             $('#networking-interfaces tr').removeClass('highlight');
             if (id) {
-                $('#networking-interfaces tr[data-interface="' + shell.esc(id) + '"]').addClass('highlight');
+                $('#networking-interfaces tr')
+                    .attr("data-interface", id)
+                    .addClass('highlight');
```
Comment (mvollmer) [inline pkg/networkmanager/interfaces.js]: The new code sets all `data-interface` attributes of all rows to `id`, no?  I agree we should stop constructing a jQuery expression here.  I think we need to explicitly loop like this

```
$('#network-interfaces tr').each(function (i, tr) {
  if ($(tr).attr('data-interface') == id)
    $(tr).addClass('highlight');
});
```

File: pkg/networkmanager/interfaces.js [hunk 2]
```diff
@@ -1396,7 +1398,7 @@ PageNetworking.prototype = {
                 var samples = usage_samples[iface];
                 var rx = samples[0][0];
                 var tx = samples[1][0];
-                var row = $('#networking-interfaces tr[data-sample-id="' + shell.esc(iface) + '"]');
+                var row = $('#networking-interfaces tr').attr("data-sample-id", iface);
```
Comment (mvollmer) [inline pkg/networkmanager/interfaces.js]: Same as above.

File: pkg/networkmanager/interfaces.js [hunk 3]
```diff
@@ -1655,7 +1657,9 @@ PageNetworkInterface.prototype = {
         function highlight_netdev_row(event, id) {
             $('#network-interface-slaves tr').removeClass('highlight');
             if (id) {
-                $('#network-interface-slaves tr[data-interface="' + shell.esc(id) + '"]').addClass('highlight');
+                $('#network-interface-slaves tr')
+                    .attr("data-interface", id)
+                    .addClass('highlight');
```
Comment (mvollmer) [inline pkg/networkmanager/interfaces.js]: Same.

File: pkg/networkmanager/interfaces.js [hunk 4]
```diff
@@ -1717,7 +1721,8 @@ PageNetworkInterface.prototype = {
                 var samples = usage_samples[iface];
                 var rx = samples[0][0];
                 var tx = samples[1][0];
-                var row = $('#network-interface-slaves tr[data-sample-id="' + shell.esc(iface) + '"]');
+                var row = $('#network-interface-slaves tr')
+                    .attr("data-sample-id", iface);
```
Comment (mvollmer) [inline pkg/networkmanager/interfaces.js]: Same.

**Comment by stefwalter** [comment]
Replaced shell.esc with encodeURIComponent(). Is that good enough?

**Comment by stefwalter** [comment]
Added encodeURIComponent() when setting data- attributes as well.

**Comment by mvollmer** [comment]
> Replaced shell.esc with encodeURIComponent(). Is that good enough?

Should be enough.

**Pull request closed without merging**